expression/generator/compare_vec.go

```diff
@@ -170,6 +170,92 @@ func (b *builtin{{ .compare.CompareName }}{{ .type.TypeName }}Sig) vectorized()
 }
 `))
 
+var builtinCoalesceCompareVecTpl = template.Must(template.New("").Parse(`
+	{{ if .type.Fixed }}
+func (b *builtin{{ .compare.CompareName }}{{ .type.TypeName }}Sig) vecEval{{ .type.TypeName }}(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	result.Resize{{ .type.TypeNameInColumn }}(n, true)
+	i64s := result.{{ .type.TypeNameInColumn }}s()
+	buf1, err := b.bufAllocator.get(types.ET{{ .type.ETName }}, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	for j := 0; j < len(b.args); j++{
+
+		if err := b.args[j].VecEval{{ .type.TypeName }}(b.ctx, input, buf1); err != nil {
+			return err
+		}
+		args := buf1.{{ .type.TypeNameInColumn }}s()
+		for i := 0; i < n; i++ {
+			if !buf1.IsNull(i) && result.IsNull(i) {
+				i64s[i] = args[i]
+				result.SetNull(i, buf1.IsNull(i))
+				continue
+			}
+			if !result.IsNull(i) {
+				continue
+			}
+			result.SetNull(i, true)
+		}
+	}
+	return nil
+}
+{{ else }}
+func (b *builtin{{ .compare.CompareName }}{{ .type.TypeName }}Sig) vecEval{{ .type.TypeName }}(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+
+	result.Reserve{{ .type.TypeNameInColumn }}(n)
+
+	buf1, err := b.bufAllocator.get(types.ET{{ .type.ETName }}, n)
```

Comment:
In most cases, `Coalesce` doesn't have many arguments, so how about allocating `len(b.args)` buffer columns at once.
Then your code would look like:
```
bufs := allocateBufs()
for i := 0; i < len(b.args); i++ {
    b.args.VecEvalString(bufs[i])
}

result.ReserveString(n)
for i := 0; i < n; i++ {
    for j := 0; j < len(b.args); j++ {
        .......
    }
}
```

Reply:
And please update the benchmark result in your PR's description after updating. @gtygo